components/public-api/gitpod/v1/workspace.proto

```diff
@@ -38,6 +132,233 @@ message WatchWorkspaceStatusResponse {
+  PaginationResponse pagination = 2;
+}
+
+message CreateAndStartWorkspaceRequest {
+  message Source {
+    string context_url = 1;
+    string repository_id = 2;
+  }
+
+  string organization_id = 1;
+
+  Source source = 2;
+  StartWorkspaceSpec spec = 3;
+
+  bool ignore_running_workspace_on_same_commit = 4;
+  bool ignore_running_prebuild = 5;
+  bool allow_using_previous_prebuilds = 6;
+  bool force_default_config = 7;
+}
+
+message CreateAndStartWorkspaceResponse { string workspace_id = 1; }
+
+message StartWorkspaceRequest { string workspace_id = 1; }
+
+message StartWorkspaceResponse {}
+
+message StopWorkspaceRequest {
+  string workspace_id = 1;
+
+  // scope of the workspace to stop
+  WorkspaceScope scope = 2;
+}
+
+message StopWorkspaceResponse {}
+
+message DeleteWorkspaceRequest { string workspace_id = 1; }
+
+message UpdateWorkspaceRequest {
+
+  string workspace_id = 1;
+
+  WorkspaceSpec spec = 2;
+}
+
+message UpdateWorkspaceResponse { Workspace workspace = 1; }
+
+message DeleteWorkspaceResponse {}
+
+message ListSupportedWorkspaceClassesRequest {}
+
+message ListSupportedWorkspaceClassesResponse {
+  repeated WorkspaceClass result = 1;
+}
+
+message GetDefaultWorkspaceImageRequest { string workspace_id = 1; }
+
+message GetDefaultWorkspaceImageResponse { string default_workspace_image = 1; }
+
+message GetWorkspaceOwnerTokenRequest { string workspace_id = 1; }
+
+message GetWorkspaceOwnerTokenResponse { string owner_token = 1; }
+
+message GetWorkspaceEditorCredentialsRequest { string workspace_id = 1; }
+
+message GetWorkspaceEditorCredentialsResponse { string editor_credentials = 1; }
+
+message SendHeartBeatRequest {
+  string workspace_id = 1;
+
+  // editor_closed indicates if the editor is closed
+  // if true, the workspace will be stopped after stopping_timeout
+  bool editor_closed = 2;
+}
+
+message SendHeartBeatResponse {}
+
+message WatchWorkspaceImageBuildLogsRequest { string workspace_id = 1; }
+
+message WatchWorkspaceImageBuildLogsResponse {
+  WorkspaceImageBuild workspace_image_build = 1;
+}
+
+message WorkspaceImageBuild {
+  enum Phase {
+    PHASE_UNSPECIFIED = 0;
+    PHASE_BASE_IMAGE = 1;
+    PHASE_GITPOD_LAYER = 2;
+    PHASE_ERROR = 3;
+    PHASE_DONE = 4;
+  }
+
+  message Log {
+    string task_id = 1;
+    string task = 2;
+    string message = 3;
+  }
+
+  Phase phase = 1;
+  string message = 2;
+  repeated Log logs = 3;
+}
+
+message ListWorkspacePortsRequest {
+  string workspace_id = 1;
+
+  PaginationRequest pagination = 2;
+}
+
+message ListWorkspacePortsResponse {
+  repeated WorkspacePort ports = 1;
+
+  PaginationResponse pagination = 2;
+}
+
+message UpdateWorkspacePortRequest {
+  string workspace_id = 1;
+
+  uint64 port = 2;
+
+  PortSpec spec = 3;
+}
+
+message UpdateWorkspacePortResponse {}
+
+message CreateWorkspaceSnapshotRequest {
+  string workspace_id = 1;
+
+  // wait indicates whether to wait for the snapshot to complete
+  bool wait = 2;
```

Comment:
i think we sohuld not have it, one shold call WaitForWorkspaceSnapshot afterwards. I believe we deprecated this property but never removed.